worker/worker.go

```diff
@@ -445,6 +462,179 @@ func (w *Worker) processClaim(claim taskClaim) {
 	}
 }
 
+// superseding returns any superseding task, and a function to be called when
+// processed to resolve other superseded tasks.
+func (w *Worker) superseding(claim taskClaim) (taskClaim, func()) {
+	// Create monitor for any problems we run into
+	m := w.monitor.WithPrefix("superseding")
+
+	var payload map[string]interface{}
+	if json.Unmarshal(claim.Task.Payload, &payload) != nil {
+		// Do nothing, if there is an error, it'll show up later and logging will be
+		// more natural.
+		return claim, func() {}
+	}
+
+	// Take supersederUrl out of the payload, as it would break the payload
+	// validation done in TaskRun. We attempt to hide superseding from the rest
+	// of the worker implementation, so that it's only creating a hack here.
```

Comment:
How come it would break the TaskRun validation? This sounds pretty hacky.

Reply:
It validates the payload against the payload schema...

And it forbids additional properties.

This is all me ensuring that these superseded properties are restricted to this file and function... So it's easy to rip out and kill this, when someday we have superseded in the queue.